HW0/P4/P4.py

```diff
@@ -40,38 +61,68 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    P4_measurements = np.loadtxt('P4_measurements.txt', delimiter=',')
+
+    x_bar = np.dot(np.diag([1.0/rx, 1.0/ry, 1.0/rz]), P4_measurements.T)
+    x_coords, y_coords, z_coords = x_bar[0,:], x_bar[1,:], x_bar[2,:]
+
+    ax.plot(x_coords, y_coords, z_coords,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.diag(np.ones(6))
+    for ii in range(3):
+        A[ii, ii+3] = dt
+        A[ii+3, ii+3] -= c * dt
+    a = np.zeros(6)
+    a[-1] = g * dt
+    s = np.zeros((6, K))
+    s[:,0] = np.array([0.0, 0.0, 2.0, 15.0, 3.5, 4.0])
 
     # Initial conditions for s0
-    # Compute the rest of sk using Eq (1)
+    # Compute the rest of  using Eq (1)
+
+    for k in xrange(1, K):
+        s[:,k] = np.dot(A, s[:, k-1]) + a
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    x_coords, y_coords, z_coords = s[0,:], s[1,:], s[2,:]
+    ax.plot(x_coords, y_coords, z_coords,
+            '-k', label='Blind trajectory')
 
     #####################
     # Part 4:
     # Use the Kalman filter for prediction
     #####################
 
-    # B = ?
-    # C = ?
+    B = np.diag([bx, by, bz, bvx, bvy, bvz])
+    C = np.zeros((3,6))
+    C[0, 0] = rx
+    C[1, 1] = ry
+    C[2, 2] = rz
 
     # Initial conditions for s0 and Sigma0
     # Compute the rest of sk using Eqs (2), (3), (4), and (5)
 
```

Comment:
I would comment on what everything is doing! (e.g. - declare the B matrix, Eq. [whatever equation it was] from the handout).